### PR TITLE
Adaptations needed after API changes

### PR DESCRIPTION
**Pull request by marcus-oscarsson**
Dear all,

Some adaptations needed after the API changes we have introduced recently, along with some code cleanup.

For Qt developers pay attention to:

                    snapshot = self.shape_history.get_scene_snapshot()
to

                    snapshot = self.shape_history.get_snapshot()

Marcus

### CODE REVIEW

File: HardwareObjects/GenericDiffractometer.py [hunk 1]
```diff
@@ -196,7 +195,7 @@ def __init__(self, name):
         self.image_height = None
 
         self.current_state = None
-        self.current_phase = None
+        self.current_phase = GenericDiffractometer.PHASE_CENTRING
```
Comment (IvarsKarpics) [inline HardwareObjects/GenericDiffractometer.py]: I would keep value None as not initialized value (all variables in __init__ should be better None) and then initiate it in the init method of current diffractometer (Mockup, etc).

File: HardwareObjects/queue_entry.py [hunk 2]
```diff
@@ -1091,7 +1095,7 @@ def collect_dc(self, dc, list_item):
                 else:
                     pos_dict = self.diffractometer_hwobj.getPositions()
                     cpos = queue_model_objects.CentredPosition(pos_dict)
-                    snapshot = self.shape_history.get_scene_snapshot()
+                    snapshot = self.shape_history.get_snapshot()
```
Comment (IvarsKarpics) [inline HardwareObjects/queue_entry.py]: is this a snapshot with or without overlays? How we can distinguish them? I guess if you call get_snapshot of shape_history you will get a snapshot with overlays and get_snapshot from camera will return a raw image.

File: HardwareObjects/GenericDiffractometer.py (hunk 1 above)
Comment (marcus-oscarsson) [reply HardwareObjects/GenericDiffractometer.py]: Yes that's probably nicer, I made a fix for it !

File: HardwareObjects/queue_entry.py (hunk 2 above)
Comment (marcus-oscarsson) [reply HardwareObjects/queue_entry.py]: Our get_snapshot method on ShapeHistory is for the moment empty, but we do have a get_snapshot that takes a overlays keyword argument.

That's another reason for moving the entire snapshot functionality along with a bunch of other things to a more beamline wide entity.

**Comment by marcus-oscarsson** [comment]
@IvarsKarpics So I can go ahead and merge this then ?

**Comment by IvarsKarpics** [comment]
Sorry, I forgot to merge it. Now its merged

**Pull request merged**